File: TimeOffManager/TimeOffManager.DataAccess/Models/TimeOff.cs
using TimeOffManager.DataAccess.Enums;

namespace TimeOffManager.DataAccess.Models
{
    public class TimeOff
    {
        public int Id { get; set; }
        public RequestType Type { get; set; }
        public DateTime ApprovedDate {  get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsApproved { get; set; }
        public string Note { get; set; }
        public int? ManagerId { get; set; }
        public Manager? Manager { get; set; }
        public int EmployeeId { get; set; }
        public Employee? Employee { get; set; }
        public int ApproverId { get; set; }
        public Manager ApproveManager { get; set; }
    }
}

File: TimeOffManager/ClientApp/timeoff-manager-client/src/models/time-off.ts
import { Manager } from "./manager";
import { User } from "./user";

export interface TimeOff {
    id:number;
    type:TimeOffType;
    approvedDate:Date;
    startDate:Date;
    endDate:Date;
    isApproved:boolean;
    note?:string;
    managerId?: number;
    manager?:Manager;
    employeeId: number;
    employee:User;
    approverId?:number;
    approveManager?:Manager;
}

export enum TimeOffType {
    Vacation = 1,
    SickLeave = 2
}